pkg/ai/amazonsagemaker.go

```diff
@@ -0,0 +1,161 @@
+/*
+Copyright 2023 The K8sGPT Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ai
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"strings"
+
+	"encoding/json"
+	"log"
+
+	"github.com/fatih/color"
+	"github.com/k8sgpt-ai/k8sgpt/pkg/cache"
+	"github.com/k8sgpt-ai/k8sgpt/pkg/util"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sagemakerruntime"
+)
+
+type SageMakerAIClient struct {
+	client      *sagemakerruntime.SageMakerRuntime
+	language    string
+	model       string
+	temperature float32
+	endpoint    string
+}
+
+const (
+	// SageMaker completion parameters
+	maxNewTokens = 256
+	top_P        = 0.9
+)
+
+type Generation struct {
+	Role    string `json:"role"`
+	Content string `json:"content"`
+}
+
+func (c *SageMakerAIClient) Configure(config IAIConfig, language string) error {
+
+	// Create a new AWS session
+	sess := session.Must(session.NewSessionWithOptions(session.Options{
+		Config:            aws.Config{Region: aws.String(config.GetProviderRegion())},
```

Comment:
I would make it clear in the README that we use the default region provided in the envs

Reply:
k8sgpt will use ProviderRegion set in config, not one from aws cli config or env, I added that info in README
https://github.com/dkuroczk/k8sgpt/blob/feat/add-amazonsagemaker/README.md?plain=1#L418